mission/functions/core/ui/taskroster/fn_tr_teamInfo_show.sqf

```diff
@@ -0,0 +1,257 @@
+	_tgHRoleLimits = _tgHRoleLimits + pixelH + UIH(0.1);
+
+};
+
+/////////////////////////////////////////////////////////////////////////////////
+// Players list with role icons
+
+//--- Delete old controls
+((allControls _displayCtrlPage) select {ctrlParentControlsGroup _x == _displayCtrlPlayerList}) apply {ctrlDelete _x};
+//--- Top seperator
+private _seperatorPlayerList = _displayCtrlPage ctrlCreate ["vn_mf_RscText", -1, _displayCtrlPlayerList];
+_seperatorPlayerList ctrlSetPosition [0, 0, UIW(15), pixelH];
+_seperatorPlayerList ctrlCommit 0;
+_seperatorPlayerList ctrlSetBackgroundColor [0,0,0,1];
+private _tgHPlayerList = pixelH + UIH(0.1); // keep track of the y position in the controls group for the next row
+
+_teamTraits apply {
+
+	private _hmap = _x;
+
+	private _addH = 0; // find the highest h value of any text control
+	private _nCtrls = []; // store newly created controls
+
+	//--- Row group
+	private _ctrlGrp = _displayCtrlPage ctrlCreate ["vn_mf_RscControlsGroupNoScrollbarHV", -1, _displayCtrlPlayerList];
+	_nCtrls pushBack _ctrlGrp;
+	_ctrlGrp ctrlSetPosition [0, _tgHPlayerList, UIW(15), UIH(1)];
+	_ctrlGrp ctrlCommit 0;
+
+	//--- player name
+	private _ctrlPlayerName = _displayCtrlPage ctrlCreate ["vn_mf_RscStructuredText", -1, _ctrlGrp];
+	_ctrlPlayerName ctrlSetPosition [0, 0, UIW(8), UIH(1)];
+	_ctrlPlayerName ctrlCommit 0;
+	_ctrlPlayerName ctrlSetStructuredText parseText format ["<t size='0.75'>%1</t>", (_hmap get "playerName")];
+	_ctrlPlayerName ctrlSetTextColor [0,0,0,1];
+
+	//--- role icons per player
+	if (count (_hmap get "traits") isEqualTo 0) then {
+		private _horizStart = 11;
+		private _ctrlIcon = _displayCtrlPage ctrlCreate ["vn_mf_RscText", -1, _ctrlGrp];
+		_nCtrls pushBack _ctrlIcon;
+		_ctrlIcon ctrlSetPosition [UIW(_horizStart), 0, UIW(3), UIH(1)];
+		_ctrlIcon ctrlCommit 0;
+		_ctrlIcon ctrlSetText "None";
+		_ctrlIcon ctrlSetTextColor [0,0,0,0.8];
+
+	} else {
+		{
+			private _horizStart = 11 + _forEachIndex;
+			private _ctrlIcon = _displayCtrlPage ctrlCreate ["vn_mf_RscPicture", -1, _ctrlGrp];
+			_nCtrls pushBack _ctrlIcon;
+			_ctrlIcon ctrlSetPosition [UIW(_horizStart), 0, UIW(1), UIH(1)];
+			_ctrlIcon ctrlCommit 0;
+			_ctrlIcon ctrlSetText (_x select 0);
+			_ctrlIcon ctrlSetTooltip (_x select 1);
+			_ctrlIcon ctrlSetTextColor [0,0,0,0.8];
+		} forEach (_hmap get "traits");
+	};
+
+	_addH = _addH max UIH(1);
+
+	_nCtrls apply {
+		_x ctrlSetPositionH _addH;
+		_x ctrlCommit 0;
+	};
+	_tgHPlayerList = _tgHPlayerList + _addH;
+
+	//--- Seperator between players
+	_seperatorPlayerList = _displayCtrlPage ctrlCreate ["RscText", -1, _displayCtrlPlayerList];
+	_seperatorPlayerList ctrlSetPosition [0, _tgHPlayerList, UIW(15), pixelH];
+	_seperatorPlayerList ctrlCommit 0;
+	_seperatorPlayerList ctrlSetBackgroundColor [0,0,0,1];
+	_tgHPlayerList = _tgHPlayerList + pixelH + UIH(0.05);
+};
+
+// right hand side -- change team listbox
+call vn_mf_fnc_tr_teamInfo_listboxInit;
```

Comment:
we could move this code in here, IF it's not called anywhere else? but then this script is already huge anyway. potentially need to break it up into some smaller scripts.

Reply:
!todo -- is it worth considering different directories for different taskroster page elements in the functions dir?